packages/edit-site/src/store/selectors.js

```diff
@@ -391,3 +391,58 @@ export const getCurrentTemplateTemplateParts = createRegistrySelector(
 export function getEditorMode( state ) {
 	return __unstableGetPreference( state, 'editorMode' );
 }
+
+/**
+ * Returns the post currently being edited in its last known saved state, not
+ * including unsaved edits. Returns an object containing relevant default post
+ * values if the post has not yet been saved.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Object} Post object.
+ */
+export const getCurrentTemplate = createRegistrySelector(
+	( select ) => ( state ) => {
+		const templateType = getEditedPostType( state );
+		const templateId = getEditedPostId( state );
+		const template = templateId
+			? select( coreDataStore ).getEntityRecord(
+					'postType',
+					templateType,
+					templateId
+			  )
+			: null;
+
+		return template;
+	}
+);
+
+/**
+ * Returns the number of revisions of the template currently being edited.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {number} Number of revisions.
+ */
+export function getCurrentTemplateRevisionsCount( state ) {
+	const template = getCurrentTemplate( state );
+	if ( template.source === 'theme' ) {
+		return 0;
+	}
+	return ( template?._links?.[ 'version-history' ]?.[ 0 ]?.count ?? 0 ) + 1;
```

Comment:
I wish we could avoid bleeding this internal API logic into the store/selectors, but I guess the `getEntityRecord()` API doesn't expose a good way to parse responses or avoid this?

Reply:
I think @dmsnell has a good point on not needing this `getCurrentTemplateRevisionsCount` selector.

Reply:
+1. We can use `template._links` directly in `useRevisionData` and avoid new selectors.

Reply:
I removed both new selectors and added the logic in the hook.